sympy/physics/continuum_mechanics/beam.py

```diff
@@ -65,34 +66,37 @@ class Beam(object):
          + Piecewise(((x - 2)**4, x - 2 > 0), (0, True))/4)/(E*I)
     """
 
-    def __init__(self, length, elastic_modulus, second_moment, variable=Symbol('x'), base_char='C'):
+    def __init__(self, length, elastic_modulus, shape, variable=Symbol('x'), base_char='C'):
         """Initializes the class.
 
         Parameters
         ==========
         length : Sympifyable
             A Symbol or value representing the Beam's length.
+
         elastic_modulus : Sympifyable
             A SymPy expression representing the Beam's Modulus of Elasticity.
             It is a measure of the stiffness of the Beam material. It can
             also be a continuous function of position along the beam.
-        second_moment : Sympifyable
-            A SymPy expression representing the Beam's Second moment of area.
-            It is a geometrical property of an area which reflects how its
-            points are distributed with respect to its neutral axis. It can
-            also be a continuous function of position along the beam.
+
+        shape : Geometry object or Sympifiable
+            Describes the shape of the beam via a Geometry object representing
+            the shape of the cross-section of the beam or via a sympy expression
+            representing the Beam's Second moment of area.
```

Comment:
You need to retain the prior explanation and add the additional describing the shape.